franklin/github/views.py

```diff
@@ -55,6 +63,152 @@ def callback(request):
+            return response
+        else:
+            logger.warn('Bad POST response code of', response.status_code)
+    return None
+
+# Not currently used. but it works.
+#def get_owners_repos(owner_name):
+#    have_next_page = True
+#    url = github_base + 'orgs/' + owner_name + '/repos?per_page=100'
+#    # TODO - Confirm that a header token is the best/most secure way to go
+#    headers = {
+#                'content-type': 'application/json',
+#                'Authorization': 'token ' + oauth
+#              }
+#    repos = []
+#
+#    while have_next_page:
+#        response = None
+#        have_next_page = False # when in doubt, we'll leave the loop after 1
+#        response = make_rest_get_call(url, headers)
+#
+#        if response is not None:
+#            # Add all of the repos to our list
+#            for repo in response.json():
+#                repo_data = {}
+#                repo_data['id'] = repo['id']
+#                repo_data['name'] = repo['name']
+#                repos.append(repo_data)
+#
+#            # If the header has a paging link called 'next', update our url
+#            # and continue with the while loop
+#            if response.links and response.links.get('next', None):
+#                url = response.links['next']['url']
+#                have_next_page = True
+#
+#    if not repos:
+#        logger.error('Failed to find repos for owner', owner_name)
+#    return repos
+
+def get_site_from_request(request):
+    owner_name = request.data.get('owner', None)
+    owner_id = request.data.get('owner_id', None)
+    repo_name = request.data.get('repository', None)
+    repo_id = request.data.get('repository_id', None)
+    if owner_id and repo_id:
+        site, created = Site.objects.get_or_create(owner_id=owner_id, 
+                                                   repo_name_id=repo_id)
+        if site:
+            site.owner = owner_name
+            site.repo_name = repo_name
+            site.oauth_token = os.environ['GITHUB_OAUTH']
+            site.save()
+            return site
+    return None
+
+
+def get_franklin_config(site):
+    url = github_base + 'repos/' + site.owner + '/' + site.repo_name + '/contents/.franklin.yml'
+    #TODO - This will fetch the file from the default master branch
+    headers = {
+                'content-type': 'application/json',
+                'Authorization': 'token ' + site.oauth_token
+              }
+    response = make_rest_get_call(url, headers)
+
+    if response is not None:
+        download_url = response.json().get('download_url', None)
+        raw_data = make_rest_get_call(download_url, None)
+        # TODO - validation and cleanup needed here similar to:
+        # http://stackoverflow.com/a/22231372
+        franklin_config = yaml.load(raw_data.text)
+        return franklin_config
+    return None
+
+def create_repo_webhook(site):
+    # TODO - check for existing webhook and update if needed (or skip)
```

Comment:
I might do this soon just because it's annoying to dev without it. You have to keep deleting the webhook manually or you'll get an error in the code.